eurocropsml/dataset/dataset.py

```diff
@@ -71,88 +78,135 @@ def __init__(
+                                    ),
+                                    1,
+                                    data[idx : idx + count],
+                                )
+                            except ValueError:
+                                raise ValueError(
+                                    "Band 4 cannot be excluded if date_type is 'months'"
+                                )
+                            month_data[month] = data[idx + np.argmin(cloud_probs)]
+                    arrays_dict["data"][satellite] = month_data
+                    arrays_dict["dates"][satellite] = np.arange(12)
         return arrays_dict
 
     def __getitem__(
         self,
         idx: int,
     ) -> LabelledData:
-        f = self.file_list[idx]
+        f = {key: paths[idx] for key, paths in self.file_dict.items()}
+
         arrays_dict = self.mmap_store[f]
 
         arrays_dict = self._format_dates(
-            self.config,
             self.preprocess_config,
             self.s2_data_bands,
             self.config.date_type,
             arrays_dict,
         )
 
-        np_data = arrays_dict.pop("data")
+        np_data_dict = arrays_dict.pop("data")
         if self.keep_band_idxs is not None:
-            np_data = np_data[:, self.keep_band_idxs]
+            np_data_dict["S2"] = np_data_dict["S2"][:, self.keep_band_idxs]
 
-        meta_data: dict[str, torch.Tensor] = {
-            array_name: torch.tensor(np_array) for array_name, np_array in arrays_dict.items()
+        meta_data: dict[str, dict[str, torch.Tensor] | torch.Tensor] = {
+            array_name: {key: torch.tensor(np_array) for key, np_array in meta_array.items()}
+            for array_name, meta_array in arrays_dict.items()
         }
+        # center is always the same, replace by first value
+        meta_data["center"] = cast(torch.Tensor, next(iter(meta_data["center"].values())))
 
-        tensor_data = torch.tensor(np_data, dtype=torch.float)
+        # normalization and scaling to (0,1]
         if self.config.normalize:
-            tensor_data = torch.mul(tensor_data, NORMALIZING_FACTOR)
+            if "S1" in np_data_dict:
+                normalized_s1 = np_data_dict["S1"] * NORMALIZING_FACTOR_S1
+                np_data_dict["S1"] = (normalized_s1 + 1) / 2 + EPSILON
+            if "S2" in np_data_dict:
+                normalized_s2 = np_data_dict["S2"] * NORMALIZING_FACTOR_S2
+                np_data_dict["S2"] = normalized_s2 * (1 - EPSILON) + EPSILON
 
         if self.pad_seq_to_366:
-            tensor_data = pad_seq_to_366(tensor_data, meta_data["dates"])
+            for satellite, value_array in np_data_dict.items():
+                np_data_dict[satellite] = pad_seq_to_366(
+                    value_array, meta_data["dates"][satellite]  # type: ignore[index]
+                )
+            meta_data["dates"] = _unique_dates(
+                cast(dict[str, torch.Tensor], meta_data["dates"]), list(f.keys())
```

Comment:
Minor: If we use Iterable instead of List for the last `_unique_dates` argument would could avoid converting to list here (and in a few more places as well). (Not a big concern though, so we could leave as it is, as these lists will always be quite short)